### PR TITLE
Kite Activation Improvements

### LINKED ISSUE #226: Extension issue
- Issue Type: `Bug`
- Extension Name: `kite`
- Extension Version: `0.82.0`
- OS Version: `Darwin x64 17.5.0`
- VSCode version: `1.34.0-insider`

:warning: We have written the needed data into your clipboard. Please paste! :warning:```json 
{
	"messages": [],
	"activationTimes": {
		"startup": true,
		"codeLoadingTime": 78,
		"activateCallTime": 7,
		"activateResolvedTime": 3,
		"activationEvent": "*"
	},
	"runtimeErrors": [
		{
			"name": "Error",
			"message": "Unable to find Kite application install using mdfind"
		},
		{
			"name": "Error",
			"message": "Unable to find Kite application install using mdfind"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'focus' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'selectionChanged' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'selectionChanged' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'selectionChanged' of undefined"
		},
		{
			"name": "TypeError",
			"message": "Cannot read property 'selectionChanged' of undefined"
		}
	]
}
```

### PR DESCRIPTION
**Pull request by edzkite**
Fixes #226 

- Add undefined checks for calls invoked by Event editors in kite.js
- Add notification if Kite Engine install cannot be found

Would love to get feedback on where notification is being emitted, and its messaging.

Screen Shot 2019-04-23 at 6 25 28 PM

### CODE REVIEW

File: src/kite.js [hunk 1]
```diff
@@ -57,6 +57,20 @@ const Kite = {
 
     Logger.LEVEL = Logger.LEVELS[vscode.workspace.getConfiguration('kite').loggingLevel.toUpperCase()];
 
+    KiteAPI.isKiteInstalled().catch(err => {
+      if (err.message.includes("Unable to find Kite application install")) {
+        vscode.window.showInformationMessage('Unable to find Kite application install.', 'Install').then(item => {
```
Comment (dbratz1177) [inline src/kite.js]: language nit: "application" => "engine" I think adds more clarity for a user about the nature of the situation

Comment (dbratz1177) [reply src/kite.js]: I wonder if adding a second sentence "The Kite engine is needed to power the completions of the Kite extention" is worthwhile too?

Comment (edzkite) [reply src/kite.js]: Yeah, I like that. I cleaned up the language a little bit, here's the latest revision.

Screen Shot 2019-04-26 at 10 54 21 AM

**Comment by dbratz1177** [comment]
dope! feel free to merge whenever

**Pull request merged**